suppressible-error-prone/src/main/java/com/palantir/suppressibleerrorprone/VisitorStateModifications.java

```diff
@@ -135,8 +136,44 @@ public static Description interceptDescription(VisitorState visitorState, Descri
                 .build();
     }
 
-    private static boolean suppressibleTree(Tree tree) {
-        return modifiersTree(tree).isPresent();
+    private static boolean suppressibleTreePath(TreePath treePath) {
+        Tree leaf = treePath.getLeaf();
+        if (!(leaf instanceof ClassTree || leaf instanceof MethodTree || leaf instanceof VariableTree)) {
+            // We can only add suppressions to classes, methods, or variables
+            return false;
+        }
+
+        if (isAnonymousClass(treePath)) {
+            // We cannot add annotations to anonymous classes
+            return false;
+        }
+
+        if (isLambdaParameter(treePath)) {
+            // We cannot add annotations to implicit lambda parameters
+            return false;
+        }
+
+        return true;
+    }
+
+    private static boolean isAnonymousClass(TreePath tree) {
+        if (!(tree.getLeaf() instanceof ClassTree classTree)) {
+            return false;
+        }
+
+        return classTree.getSimpleName().isEmpty();
+    }
+
+    private static boolean isLambdaParameter(TreePath tree) {
+        if (!(tree.getLeaf() instanceof VariableTree variableTree)) {
+            return false;
+        }
+
+        if (!(tree.getParentPath().getLeaf() instanceof LambdaExpressionTree lambdaExpressionTree)) {
+            return false;
+        }
+
+        return lambdaExpressionTree.getParameters().contains(variableTree);
```

Comment:
Nice! Let's add this to `SuppressWarningsUtils`? I have a use case for this in a future PR : )